client/asset/dcr/dcr.go

```diff
@@ -2538,8 +2538,10 @@ func (dcr *ExchangeWallet) ValidateSecret(secret, secretHash []byte) bool {
 // the specified swap. The contract and matchTime are provided so that wallets
 // may search for the coin using light filters.
 //
-// If the swap was not funded by this wallet, and it is already spent, this
-// method may return asset.CoinNotFoundError. Compare dcr.externalTxOut.
+// For a non-SPV wallet, if the swap appears spent but it cannot be located in a
+// block with a cfilters scan, this will return asset.CoinNotFoundError. For SPV
```

Comment:
Thinking about this again, it can't appear spent and not be located in a block. Not currently anyways. Per the code docs on dcrd's gettxout handler, dcrd treats an output as unspent (whether mined or unmined) as long as it isn't spent in a mined transaction. So if gettxout returns a nil response and the output isn't found by a block search, it's most likely in mempool, just not the node's mempool.

Reply:
> . So if gettxout returns a nil response and the output isn't found by a block search, it's most likely in mempool, just not the node's mempool

Either that or the tx really doesn't exist.  Either way, it's cause for concern.  A full node should pick up on it in short order and if it doesn't then there is connectivity trouble.  We've seen it happen like that

Reply:
Right, I meant regarding the expression "if the swap appears spent but it cannot be located in a block with a cfilters scan" for non-spv wallets. It seems to me that a full node not being able to locate the swap in a block search means it's not spent. Maybe I'm not fully understanding what you meant by that line of comment.

Maybe... `For a non-SPV wallet, if the swap isn't found by the wallet and it cannot be located in block with a cfilters scan, ...`.

Reply:
Ah, yes the "appears spent" wording is confusing, but that's just referring to the `gettxout` part where a nil indicates *not unspent* (whether the output exists or not).  Will rephrase this.